src/sources/kafka.rs

```diff
@@ -218,11 +233,67 @@ impl SourceConfig for KafkaSourceConfig {
             cx.shutdown,
             cx.out,
             acknowledgements,
+            log_namespace,
         )))
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(self.decoding.output_type())]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        let log_namespace = global_log_namespace.merge(self.log_namespace);
+        let keys = self.keys();
+
+        let schema_definition = self
+            .decoding
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(keys.timestamp))),
+                &owned_value_path!("timestamp"),
+                Kind::timestamp(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(keys.topic))),
+                &owned_value_path!(default_topic_key().as_str()),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(keys.partition))),
+                &owned_value_path!(default_partition_key().as_str()),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(keys.offset))),
+                &owned_value_path!(default_offset_key().as_str()),
+                Kind::bytes(),
+                None,
+            )
+            .with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(keys.headers))),
+                &owned_value_path!(default_headers_key().as_str()),
+                Kind::object(Collection::empty().with_unknown(Kind::bytes())),
+                None,
+            );
+
+        let schema_definition = if let LogNamespace::Legacy = log_namespace {
+            schema_definition.with_source_metadata(
+                Self::NAME,
+                Some(LegacyKey::Overwrite(owned_value_path!(keys.key_field))),
+                &owned_value_path!(default_key_field().as_str()),
+                Kind::bytes(),
+                None,
+            )
+        } else {
```

Comment:
This jumps out to me as odd? `with_source_metadata()` handles the insertion between legacy and vector namespaces, why are we only doing this on the Legacy side? Do we not insert values into the `key_field` in the new namespace?

Reply:
I added that block because the schema field `"message_key"` was making it into vector ns's. Looking at the logic for `with_source_metadata`, the schema change in vector ns's is always attempted, since the vector path isn't an `Option`.

This is a bit different than other PRs because of this `keys` struct that's used, which always has `key_field` set. If it was an `Option`, then I wouldn't need the `if let` block.

Reply:
Yep, I actually misinterpreted what this field was for. We actually need that field in all NS's. Good callout!